optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/util/PromiseUtils.java

```diff
@@ -16,11 +16,53 @@
 
 package org.optaplanner.openshift.employeerostering.gwtui.client.util;
 
+import javax.inject.Singleton;
+
+import com.google.gwt.core.client.GWT;
+import elemental2.core.Error;
+import elemental2.core.JsObject;
 import elemental2.promise.Promise;
+import elemental2.promise.Promise.CatchOnRejectedCallbackFn;
+import elemental2.promise.Promise.PromiseExecutorCallbackFn;
 
+@Singleton
 public class PromiseUtils {
 
-    public static Promise<Void> resolve() {
+    public Promise<Void> resolve() {
         return Promise.resolve((Promise<Void>) null);
     }
+
+    public <T> Promise<T> promise(PromiseExecutorCallbackFn<T> executor) {
+        return manage(new Promise<>(executor));
+    }
+
+    public <T> Promise<T> manage(Promise<T> promise) {
+        return promise.catch_(this.<T> getDefaultCatch());
+    }
```

Comment:
I know i kinda suggested it but now I'm starting to think that "manage" is not a good name. We're creating a promise with a fixed catch statement which always returns a resolved promise. Maybe it's not a good idea to do that. We can hand the exception over to GWT but maybe have it reject anyways.. wdyt @Christopher-Chianelli?

Reply:
I tried rejecting it like `return new Promise<>((res,rej) -> rej.onInvoke(e));`, but that created an infinite loop on `getDefaultCatch()`. Maybe `attempt` for the new name?

Reply:
I think it's a good option.